ci/tests/general/general_alba.py

```diff
@@ -630,36 +630,56 @@ def filter_disks(disk_names, amount, disk_type):
         :param disk_type: Type of disk
         :return: Filtered disks
         """
-        count = 0
         node_ids = []
-        filtered_disks = dict()
+        list_of_available_disks = {}
+        filtered_disks = {}
+        disk_count = 0
+        # disk_names = dictionary with node_ids as keys and values as a list of uninitialised disk names
         for node_id in disk_names.iterkeys():
             node_ids.append(node_id)
-        while count < amount:
-            node_id = node_ids[count % len(node_ids)]
-
+            list_of_available_disks[node_id] = []
+            filtered_disks[node_id] = []
             alba_node = AlbaNodeList.get_albanode_by_node_id(node_id)
             storagerouter = GeneralStorageRouter.get_storage_router_by_ip(ip=alba_node.ip)
             root_client = SSHClient(storagerouter, username='root')
             hdds, ssds = GeneralDisk.get_physical_disks(client=root_client)
-
             if disk_type == 'SATA':
-                list_to_check = hdds.values()
+                for hdd in hdds.values():
+                    # add it to list_of_available_disks only if it's found in the unitialised list for that node
+                    if hdd['name'] in disk_names[node_id]:
+                        list_of_available_disks[node_id].append(hdd)
             elif disk_type == 'SSD':
-                list_to_check = ssds.values()
+                for ssd in ssds.values():
+                    # add it to list_of_available_disks only if it's found in the unitialised list for that node
+                    if ssd['name'] in disk_names[node_id]:
+                        list_of_available_disks[node_id].append(ssd)
+            else:
+                for hdd in hdds.values():
```

Comment:
I'd use disk_type=None in the functions signature to get in this else construct
no we always specify the type I think ... so we'll never reach this one
worth testing this part also